java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -212,11 +215,20 @@ public GeneratorResult generateJava() {
         if (!prop.isNullable) {
           // from https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable :
           // "_null values for fields that either don't specify the nullable flag, or give it a false
-          // value, will be pruned before defaulting happens..._"
+          // value, will be pruned before defaulting happens. If a default is present, it will be applied_"
           objField.addAnnotation(
               new SingleMemberAnnotationExpr(
                   new Name("com.fasterxml.jackson.annotation.JsonSetter"),
                   new NameExpr("nulls = com.fasterxml.jackson.annotation.Nulls.SKIP")));
+          if (prop.defaultValue != null) {
+            // always add an "init_*" method which will be expanded by statements needed to create the default value
```

Comment:
The actual body of the method is trivial, I would suggest to avoid generating additional methods that can possibly cause collisions in edge cases. Can you directly assign the block expression generated by `generateDefaultInitializerBody` ?

Reply:
Indeed, fixed.